File: android/src/main/java/com/margelo/nitro/nitrovideo/enums/AudioMixingMode.kt
package com.margelo.nitro.nitrovideo.enums

enum class AudioMixingMode(val value: String) {
    MIX_WITH_OTHERS("mixWithOthers"),
    DUCK_OTHERS("duckOthers"),
    AUTO("auto"),
    DO_NOT_MIX("doNotMix");

    val priority: Int
        get() = when (this) {
            DO_NOT_MIX -> 3
            AUTO -> 2
            DUCK_OTHERS -> 1
            MIX_WITH_OTHERS -> 0
        }

    companion object {
        fun fromString(str: String?): AudioMixingMode {
            return entries.firstOrNull { it.value.equals(str, ignoreCase = true) } ?: AUTO
        }
    }
}

File: ios/Enums/AudioMixingMode.swift
import AVFoundation
import Foundation

public extension AudioMixingMode {
  func priority() -> Int {
    switch self {
    case .donotmix:
      return 3
    case .auto:
      return 2
    case .duckothers:
      return 1
    case .mixwithothers:
      return 0
    }
  }

  func toSessionCategoryOption() -> AVAudioSession.CategoryOptions? {
    switch self {
    case .duckothers:
      return .duckOthers
    case .mixwithothers:
      return .mixWithOthers
    case .donotmix, .auto:
      return nil
    }
  }
}
